.evergreen/install-cli.sh

```diff
@@ -27,31 +28,55 @@ if [ "${DOCKER_RUNNING:-}" == "true" ]; then
   UV_TOOL_DIR=$_root_dir/uv-tool
 fi
 
-. ./venv-utils.sh
+# If uv is not on path, try see if it is available from the Python toolchain.
+if ! command -v uv >/dev/null; then
+  _bin_path=""
+  if [ "Windows_NT" == "${OS:-}" ]; then
+    _bin_path="/cygdrive/c/Python/Current"
+  elif [ "$(uname -s)" != "Darwin" ]; then
+    _bin_path="/Library/Frameworks/Python.Framework/Versions/Current"
+  else
+    _bin_path="/opt/python/Current"
+  fi
+  if [ -d "${_bin_path}" ]; then
+    export PATH="${_bin_path}:$PATH"
+  fi
+fi
 
-if [ ! -d $SCRIPT_DIR/venv ]; then
+# If uv is still not available, we need a venv.
+if ! command -v uv >/dev/null; then
+  . ./venv-utils.sh
 
-  . ./find-python3.sh
+  if [ ! -d $SCRIPT_DIR/venv ]; then
 
-  echo "Ensuring python binary..."
-  PYTHON=$(ensure_python3 2>/dev/null)
-  echo "Ensuring python binary... done."
+    . ./find-python3.sh
 
-  echo "Creating virtual environment 'venv'..."
-  venvcreate "${PYTHON:?}" venv
-  echo "Creating virtual environment 'venv'... done."
-else
-  venvactivate venv
+    echo "Ensuring python binary..."
+    PYTHON=$(ensure_python3 2>/dev/null)
+    echo "Ensuring python binary... done."
+
+    echo "Creating virtual environment 'venv'..."
+    venvcreate "${PYTHON:?}" venv
+    echo "Creating virtual environment 'venv'... done."
+  else
+    venvactivate venv
+  fi
 fi
 
+# If uv is still not available, we need to install it into the venv.
 if ! command -v uv >/dev/null; then
   UV_UNMANAGED_INSTALL=1 python -m pip install -q --force-reinstall uv
 fi
 
 command -V uv # Ensure a working uv binary is present.
 
+# Ensure there is a venv available for backwards compatibility.
+if [ ! -d venv ]; then
+  uv venv venv
+fi
+
 # Store paths to binaries for use outside of current working directory.
-python_binary="$(which python)"
+python_binary="$(uv run python -c 'import sys;print(sys.executable)')"
```

Comment:
```suggestion
python_binary="$(uv run --no-project python -c 'import sys;print(sys.executable)')"
```

Avoid inadvertently detecting and using a project configuration at this point.